bayesian/api_v1.py

```diff
@@ -1217,18 +1217,34 @@ class CveByDateEcosystem(Resource):
     method_decorators = [login_required]
 
     @staticmethod
-    def get(modified_date, ecosystem=None):
+    def get(end_date, ecosystem='all'):
         """Implement GET Method."""
-        if not modified_date:
+        if not end_date:
             raise HTTPError(400, error="Expected date in the request")
         try:
-            datetime.datetime.strptime(modified_date, '%Y%m%d')
+            datetime.datetime.strptime(end_date, '%Y%m%d')
         except ValueError:
             msg = 'Invalid datetime specified. Please specify in YYYYMMDD format'
             raise HTTPError(400, msg)
-        getcve = CveByDateEcosystemUtils(modified_date, ecosystem)
-        result = getcve.get_cves_by_date() if not ecosystem else getcve.get_cves_by_date_ecosystem()
-        return jsonify(result), 200
+        # Request Parameters Used
+        result_type = request.args.get('result_type', 'epvs')
+        current_app.logger.info(f'result type received:{result_type}')
+
+        if result_type == 'cves':
+            cve_sources = request.args.get('cve_sources', 'all')
+            date_range = request.args.get('date_range', '7')
```

Comment:
Here, if you don't provide date_range, you are taking date_range as non-numeric 7. That will fail at the next step

Reply:
You can also type convert to something like
```
try:
    date_range = int(request.args.get('date_range', 7))
except TypeError:
    raise HTTPError(somemessage)
```

Reply:
Actually, I am converting the non-numeric value into int value before calling the constructor. So,It will not fail. But, your suggestion also sounds as a good alternative. Will check that and make change accordingly.